x-pack/legacy/plugins/siem/public/components/open_timeline/export_timeline/index.tsx

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useState, useCallback, Dispatch, SetStateAction } from 'react';
+import { OpenTimelineResult, DeleteTimelines } from '../types';
+import { ExportSelectedData } from '../../generic_downloader';
+
+import { TimelineDownloader } from './export_timeline';
+import { DeleteTimelineModalOverlay } from '../delete_timeline_modal';
+import { exportSelectedTimeline } from '../../../containers/timeline/all/api';
+
+export interface ExportTimeline {
+  disableExportTimelineDownloader: () => void;
+  enableExportTimelineDownloader: () => void;
+  exportedIds: string[] | undefined;
+  isEnableDownloader: boolean;
+  setIsEnableDownloader: Dispatch<SetStateAction<boolean>>;
+}
+
+export const useExportTimeline = ({
+  selectedItems,
+  setActionTimeline,
+}: {
+  selectedItems?: OpenTimelineResult | OpenTimelineResult[];
```

Comment:
I think to just always be array of OpenTimelineResult will simplify some code, it is ok to only have one item